### ISSUE TITLE
Regression `used-before-assignment` with type annotation in conditional

### ISSUE THREAD

**Issue by whonore** [PyCQA/pylint#5499]
### Bug description

```py
import random

class A:
    def f(self) -> None:
        if random.random() < 0.5:
            _x: A = self  # false-positive: used-before-assignment

    def g(self) -> None:
        _x: A = self  # no warning
```

This is the same issue reported and fixed in #5342 but it still occurs if the assignment is inside a conditional.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
test.py:6:16: E0601: Using variable 'A' before assignment (used-before-assignment)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.9.6 (default, Aug 16 2021, 15:13:47) 
[GCC 9.3.0]
```
